cli/src/commit_templater.rs

```diff
@@ -41,11 +42,34 @@ use crate::templater::{
 };
 use crate::text_util;
 
+/// Trait for extending the commit template language.
+pub trait CommitTemplateLanguageExtension {
+    /// Returns a new template property on successful parse.
+    /// Else, it must return `property` back as an error.
+    fn build_commit_property_opt<'repo>(
+        &self,
+        property: Box<dyn TemplateProperty<Commit, Output = Commit> + 'repo>,
+        name: &str,
+    ) -> Result<
+        CommitTemplatePropertyKind<'repo>,
+        Box<dyn TemplateProperty<Commit, Output = Commit> + 'repo>,
+    >;
```

Comment:
In order to provide better error messages (like "Did you mean x?"), big `match` statements in `build_` functions will have to be changed to `HashMap<str, fn(..)>` similar to `revset::BUILTIN_FUNCTION_MAP`. And extension methods will have to be conceptually merged in to that table.
https://github.com/martinvonz/jj/blob/8e4d1af98eccf1ed8a79a7ddf9500467aa7684f5/lib/src/revset.rs#L1129-L1326

I don't know how nice the extension API should be abstracted, but under the hood, each extension method is `(name, fn)` pair, and name resolution will be made by the core templater, not by the extension.

Reply:
> big `match` statements in `build_` functions will have to be changed to `HashMap<str, fn(..)>`

This is still WIP, but I expect the keywords table would look like
https://github.com/yuja/jj/commit/802ccd2075d478c6a86d10be9ff55fb8db42be45

Reply:
> > big `match` statements in `build_` functions will have to be changed to `HashMap<str, fn(..)>`
> 
> This is still WIP, but I expect the keywords table would look like [yuja@802ccd2](https://github.com/yuja/jj/commit/802ccd2075d478c6a86d10be9ff55fb8db42be45)

Do you want to push that separately first?  I can patch it and finish it up in this PR otherwise

Reply:
If extension support isn't urgent, can you wait a couple of days? I'm doing some cleanup (to merge keywords into commit methods table, etc.)

Reply:
Sounds good.  I'm taking a long weekend starting tomorrow anyway so I won't be back at this until Wednesday